### PR TITLE
[SYCL] Rename project to oneAPI DPC++ Compiler

### PR DESCRIPTION
**Pull request by pvchupin**
Rename project and fix docs accordingly to new project name.
Rename some of the documentation files.
Minor docs cleanup.

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -70,10 +70,11 @@ commit automatically with `git commit -s`.
...
 
 - Create a personal fork of the project on GitHub
-- Use **sycl** branch as baseline for your changes
+  - For DPC++ Compiler project use **sycl** branch as baseline for your changes
 - Prepare your patch (follow
   [LLVM coding standards](https://llvm.org/docs/CodingStandards.html))
-- Build the project and run all tests (see
+- Build the project and run all tests
+  - For DPC++ Compiler project (see
 [GetStartedWithSYCLCompiler.md](sycl/doc/GetStartedWithSYCLCompiler.md))
```
Comment (AGindinson) [inline CONTRIBUTING.md]: Should this file be renamed as well?

File: sycl/doc/FAQ.md [hunk 2]
```diff
@@ -2,51 +2,51 @@
...
 
 
-## Developing with SYCL
+## Developing with DPC++
 
-### Q: What do I need to start developing with SYCL?
-**A:** To get the full SYCL experience you need a SYCL-capable compiler. Intel
-SYCL compiler provides you with both host and device side compilation. Another
+### Q: What do I need to start developing with DPC++?
+**A:** To get the full DPC++ experience you need DPC++ compiler. oneAPI
```
Comment (AGindinson) [inline sycl/doc/FAQ.md]:
```suggestion
**A:** To get the full DPC++ experience you need the DPC++ compiler. oneAPI
```
or maybe
```suggestion
**A:** To get the full DPC++ experience, oneAPI DPC++ compiler is required. The
```

File: CONTRIBUTING.md (hunk 1 above)
Comment (pvchupin) [reply CONTRIBUTING.md]: Thanks. Missed.

File: sycl/doc/FAQ.md (hunk 2 above)
Comment (pvchupin) [reply sycl/doc/FAQ.md]: Fixed

File: README.md [hunk 3]
```diff
@@ -4,21 +4,20 @@
 
 Intel staging area for llvm.org contribution.
 Home for Intel LLVM-based projects:
- - SYCL* Compiler and Runtimes - compiler and runtime libraries for SYCL ([https://www.khronos.org/sycl/](https://www.khronos.org/sycl/)). See **sycl** branch.
+ - oneAPI Data Parallel C++ compiler - see **sycl** branch. More information on oneAPI and DPC++ is available at ([https://www.oneapi.com/](https://www.oneapi.com/))
```
Comment (bader) [inline README.md]: Could you re-format the text in this document to keep 80-character limit, please?

File: README.md [hunk 4]
```diff
@@ -4,21 +4,20 @@
...
 
 ## Sub-projects Documentation
- - SYCL Compiler and Runtimes - See [GetStartedWithSYCLCompiler.md](sycl/doc/GetStartedWithSYCLCompiler.md)
+ - oneAPI Data Parallel C++ compiler - See [GetStartedGuide.md](sycl/doc/GetStartedGuide.md)
 
-*Other names and brands may be claimed as the property of others.
+## DPC++ extensions
 
-## SYCL Extension Proposal Documents
+DPC++ is an open, cross-architecture language built upon the ISO C++ and Khronos SYCL* standards. DPC++ extends these standards with a few extensions. See [sycl/doc/extensions](sycl/doc/extensions) for more details.
```
Comment (bader) [inline README.md]: I think we might have more than "a few" extensions, so maybe it's better to rephrase to something like:
```suggestion
DPC++ extends these standards with a number of extensions, which can be found in [sycl/doc/extensions](sycl/doc/extensions) directory.
```

File: sycl/doc/PluginInterface.md [hunk 5]
```diff
@@ -113,7 +113,7 @@ in a data section.
 
 ### The Interoperability PI APIs
 
-These are APIs needed to implement SYCL runtime interoperability with underlying
+These are APIs needed to implement DPC++ runtime interoperability with underlying
```
Comment (bader) [inline sycl/doc/PluginInterface.md]: Please, keep 80-character limit.

File: sycl/doc/GetStartedGuide.md [hunk 6]
```diff
@@ -259,27 +258,28 @@ ninja check-all
...
 skipped.
 
-### Run Khronos SYCL conformance test suite (optional)
+### Run Khronos\* SYCL\* conformance test suite (optional)
 
-Khronos SYCL conformance test suite (CTS) is intended to validate SYCL
-implementation conformance to Khronos SYCL specification.
+Khronos\* SYCL\* conformance test suite (CTS) is intended to validate implementation
+conformance to Khronos\* SYCL\* specification. DPC++ compiler is expected to pass
+significant number of tests, and it keeps improving.
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md]: Please, keep 80-character limit.

File: sycl/doc/GetStartedGuide.md [hunk 7]
```diff
@@ -479,26 +479,31 @@ class CUDASelector : public cl::sycl::device_selector {
...
 
-- SYCL device compiler fails if the same kernel was used in different
+- DPC++ device compiler fails if the same kernel was used in different
   translation units.
 - SYCL host device is not fully supported.
 - 32-bit host/target is not supported.
-- SYCL works only with OpenCL implementations supporting out-of-order queues.
-- On Windows linking SYCL applications with `/MTd` flag is known to cause crashes.
+- DPC++ works only with OpenCL low level runtimes which support out-of-order queues.
+- On Windows linking DPC++ applications with `/MTd` flag is known to cause crashes.
```
Comment (bader) [inline sycl/doc/GetStartedGuide.md]: Please, keep 80-character limit.

File: sycl/doc/FAQ.md [hunk 8]
```diff
@@ -119,9 +119,9 @@ specification.
 
 ## Device specific questions and issues
 
-### Q: What devices are supported by Intel SYCL compiler?
-**A:** By design, SYCL is closely connected to OpenCL, which is used to offload
-code to specialized devices. Intel SYCL compiler currently makes use of SPIR-V,
+### Q: What devices are supported by DPC++ compiler?
+**A:** By design, DPC++ and SYCL are closely connected to OpenCL, which is used
+to offload code to specialized devices. DPC++ compiler currently makes use of SPIR-V,
```
Comment (bader) [inline sycl/doc/FAQ.md]: Please, keep 80-character limit.
A few place in this document.

File: README.md (hunk 3 above)
Comment (pvchupin) [reply README.md]: np

File: README.md (hunk 4 above)
Comment (pvchupin) [reply README.md]: Looks better indeed. Thanks.

File: sycl/doc/PluginInterface.md (hunk 5 above)
Comment (pvchupin) [reply sycl/doc/PluginInterface.md]: done

File: sycl/doc/GetStartedGuide.md (hunk 6 above)
Comment (pvchupin) [reply sycl/doc/GetStartedGuide.md]: done

File: sycl/doc/GetStartedGuide.md (hunk 7 above)
Comment (pvchupin) [reply sycl/doc/GetStartedGuide.md]: done

File: sycl/doc/FAQ.md (hunk 8 above)
Comment (pvchupin) [reply sycl/doc/FAQ.md]: done, but kept headers and dumps untouched.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -70,11 +70,12 @@ commit automatically with `git commit -s`.
 ### Development
 
 - Create a personal fork of the project on GitHub
-- Use **sycl** branch as baseline for your changes
+  - For DPC++ Compiler project use **sycl** branch as baseline for your changes
```
Comment (keryell) [inline CONTRIBUTING.md]:
```suggestion
  - For the DPC++ SYCL Compiler project, use **sycl** branch as baseline for your changes
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -70,11 +70,12 @@ commit automatically with `git commit -s`.
...
 - Prepare your patch (follow
   [LLVM coding standards](https://llvm.org/docs/CodingStandards.html))
-- Build the project and run all tests (see
-[GetStartedWithSYCLCompiler.md](sycl/doc/GetStartedWithSYCLCompiler.md))
+- Build the project and run all tests
+  - For DPC++ Compiler project (see
```
Comment (keryell) [inline CONTRIBUTING.md]: Not clear what is the value of adding a subitem in the middle.

File: sycl/doc/CompilerAndRuntimeDesign.md [hunk 11]
```diff
@@ -78,7 +78,7 @@ Q.submit([&](handler& cgh) {
 ...
 ```
 
-In this example, the SYCL compiler needs to compile the lambda expression passed
+In this example, the compiler needs to compile the lambda expression passed
```
Comment (keryell) [inline sycl/doc/CompilerAndRuntimeDesign.md]: Is there some real value to remove SYCL everywhere from here?

File: sycl/doc/CompilerAndRuntimeDesign.md [hunk 12]
```diff
@@ -160,7 +160,7 @@ must be passed to the clang driver:
 
 `-fsycl`
 
-With this option specified, the driver will invoke the host SYCL compiler and a
+With this option specified, the driver will invoke the host compiler and a
 number of device compilers for targets specified in the `-fsycl-targets`
```
Comment (keryell) [inline sycl/doc/CompilerAndRuntimeDesign.md]: Instead of removing it for the host you could add this :-)
```suggestion
number of SYCL device compilers for targets specified in the `-fsycl-targets`
```

File: sycl/doc/CompilerAndRuntimeDesign.md [hunk 13]
```diff
@@ -546,17 +546,11 @@ compiler:
 
 ### Compiler/Runtime interface
 
-## SYCL Runtime architecture
+## DPC++ Runtime architecture
 
 *TBD*
 
-## Supported extensions
+## DPC++ Language extensions
```
Comment (keryell) [inline sycl/doc/CompilerAndRuntimeDesign.md]:
```suggestion
## DPC++ Language extensions to SYCL
```

File: sycl/doc/FAQ.md [hunk 14]
```diff
@@ -2,51 +2,51 @@
...
 
 
-## Developing with SYCL
+## Developing with DPC++
 
-### Q: What do I need to start developing with SYCL?
-**A:** To get the full SYCL experience you need a SYCL-capable compiler. Intel
-SYCL compiler provides you with both host and device side compilation. Another
+### Q: What do I need to start developing with DPC++?
+**A:** To get the full DPC++ experience you need oneAPI DPC++ compiler. DPC++
```
Comment (keryell) [inline sycl/doc/FAQ.md]:
```suggestion
**A:** To get the full DPC++ experience you need the unique oneAPI DPC++ compiler. DPC++
```
It looks quite less generic this way :-)

File: sycl/doc/GetStartedGuide.md [hunk 15]
```diff
@@ -31,82 +30,85 @@ OpenCL&trade; API to offload computations to accelerators.
   * Windows: `Visual Studio` version 15.7 preview 4 or later -
     https://visualstudio.microsoft.com/downloads/
 
-## Create SYCL workspace
+## Create DPC++ workspace
 
-Throughout this document `SYCL_HOME` denotes the path to the local directory
-created as SYCL workspace. It might be useful to create an environment variable
+Throughout this document `DPCPP_HOME` denotes the path to the local directory
```
Comment (keryell) [inline sycl/doc/GetStartedGuide.md]: This is a typical example where `DPCPP_HOME` is a good name that makes sense since it is related to this implementation.

File: sycl/doc/GetStartedGuide.md [hunk 16]
```diff
@@ -400,14 +404,14 @@ if clang was built with the cmake option `SYCL_BUILD_PI_CUDA=ON`.
...
-must be selected at runtime using the `SYCL_BE` environment variable. 
+Currently, when the application has been built with the CUDA target, the CUDA
+backend must be selected at runtime using the `SYCL_BE` environment variable.
 
 ```bash
 SYCL_BE=PI_CUDA ./simple-sycl-app-cuda.exe
 ```
 
-NOTE: SYCL developer can specify SYCL device for execution using device
+NOTE: DPC++/SYCL developer can specify SYCL device for execution using device
```
Comment (keryell) [inline sycl/doc/GetStartedGuide.md]: Voilà ! :-)

**Comment by pvchupin** [comment]
Thanks Ronan for detailed review!!!
Like you mentioned the intention of this change is to start calling project what it really is matching naming used currently in many other places (sites, forums, presentations, etc.). There is some confusion around which I hope will be fixed by this PR. Before we finalized DPC++ name we have been calling some implementation specific details inconsistently. 
I've tried to be careful doing all replacements in this change and keep SYCL everywhere when referring to SYCL standard and SYCL language concepts. Thank you for suggesting a few corrections which I really like!

I'd like to make a few (I hope obvious) statements:
* For the foreseeable future SYCL standard will be the core of DPC++.
* We plan to stay aligned with new SYCL standards development and keep supporting new standard versions as they arrive.
* We plan to keep experimenting and implementing some innovative ideas using this project as an engineering sandbox, we believe prototyping is essential for success of any language extension we'd like to explore.
* We will continue to contribute the most successful or promising extensions to the SYCL standard.
* We will continue to work on compiler upstream to llvm.org, it's a long path to land SYCL core implementation and we started from the SYCL core.

Basically our plans are not changing with this PR.

File: CONTRIBUTING.md (hunk 9 above)
Comment (pvchupin) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 10 above)
Comment (pvchupin) [reply CONTRIBUTING.md]: There could be more projects hosted in the future at intel/llvm, that's why all these subitems are there. I updated text a bit.

File: sycl/doc/CompilerAndRuntimeDesign.md (hunk 11 above)
Comment (pvchupin) [reply sycl/doc/CompilerAndRuntimeDesign.md]: In a few places like that (when it can be applied well enough to any other SYCL/DPC++ compiler or ) I just tried to be more generic by saying just compiler.

File: sycl/doc/CompilerAndRuntimeDesign.md (hunk 12 above)
Comment (pvchupin) [reply sycl/doc/CompilerAndRuntimeDesign.md]: *SYCL* device compiler is a good addition (referring to SYCL device concept), for host it's not so obvious, imho host is just host everywhere :)

File: sycl/doc/CompilerAndRuntimeDesign.md (hunk 13 above)
Comment (pvchupin) [reply sycl/doc/CompilerAndRuntimeDesign.md]: Thanks! Like it!

File: sycl/doc/FAQ.md (hunk 14 above)
Comment (pvchupin) [reply sycl/doc/FAQ.md]: We are not so paranoid about our uniqueness :) After all everybody can fork our compiler and try something on top. Keep calling it DPC++ in that case might make sense since DPC++ extensions are there.

File: sycl/doc/GetStartedGuide.md (hunk 15 above)
Comment (pvchupin) [reply sycl/doc/GetStartedGuide.md]: Thanks )

File: sycl/doc/GetStartedGuide.md (hunk 16 above)
Comment (pvchupin) [reply sycl/doc/GetStartedGuide.md]: I generally don't like slashes in naming, they add confusion most of the times.
In this particular case I guess I wanted to recognize developers who want to use only core SYCL standard features. Now looking into the sentence I'm thinking that saying just "developer" could look better :) Anyway, this is also fine.

File: CONTRIBUTING.md [hunk 17]
```diff
@@ -70,11 +70,11 @@ commit automatically with `git commit -s`.
 ### Development
 
 - Create a personal fork of the project on GitHub
-- Use **sycl** branch as baseline for your changes
+  - For the DPC++ Compiler project, use **sycl** branch as baseline for your
+    changes. See [Get Started Guide](sycl/doc/GetStartedGuide.md).
 - Prepare your patch (follow
   [LLVM coding standards](https://llvm.org/docs/CodingStandards.html))
-- Build the project and run all tests (see
-[GetStartedWithSYCLCompiler.md](sycl/doc/GetStartedWithSYCLCompiler.md))
```
Comment (AlexeySachkov) [inline CONTRIBUTING.md]: Note: you have removed a link to the corresponding docs. Was it intended?

Comment (pvchupin) [reply CONTRIBUTING.md]: I actually moved it few lines above (under new name, because I renamed file).

**Comment by keryell** [comment]
> Basically our plans are not changing with this PR.

Thank you for all the clarifications.
